### DISCUSSION TITLE
can you read svg from buffer?

### DISCUSSION THREAD

**Discussion by Pshemas** [flet-dev/flet#4243] [Q&A]
### Question

Is it possible to read and display svg image from buffer?

I'd like to generatate some images (qr codes for example) and I really don't want to save them as a regular file on disk. With TkInter etc I could simply add them to `BytesIO()` and pass it to appropriate widget. I couldn't reproduce it with Flet though.
The best result so far was with PNG, where I could b64encode the contents and buffer and use it with `src_base64` parameter.

Is there a way to avoid that and pass the genereted image directly? Can you make it work with SVG?

### Code sample

```python
# sample with qrcode library

import flet as ft
from io import BytesIO
import base64
import qrcode


# this is a bit ugly, but works with PNG (default in qrcode)
def main(page: ft.Page):
    page.add(ft.SafeArea(ft.Text("Hello, Flet!")))
    img = qrcode.make("Some text")
    buffer = BytesIO()
    img.save(buffer)
    qr_img_b64 = base64.b64encode(buffer.getvalue()).decode()
    page.add(ft.Image(src_base64=qr_img_b64, width=400, height=400))


# can you make it work with SVG?


ft.app(main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Can you share an svg base64 here ? Or runnable code you use to generate it.
